### ISSUE TITLE
Installation issue on Mac OS X (Yosemite)

### ISSUE THREAD

**Issue by doggan** [lovell/sharp#228]
Hello. I'm trying to install the sharp package.

I'm using node v0.10.35 and Xcode 6.3.1.

I installed libvips via: ```brew install homebrew/science/vips --with-webp --with-graphicsmagick```

I confirmed libvips is installed:
```bash
$ vips --version
vips-8.0.2-Sat May 30 17:54:55 PDT 2015
```

Then I attempt to install sharp via npm:
```bash
$ npm install sharp
npm WARN package.json recursive-image-search@0.1.0 No description
\
> sharp@0.10.0 install /Users/shyam/Documents/_dev/recursive-image-search/node_modules/sharp
> node-gyp rebuild

2015-05-30 18:20:08.012 xcodebuild[2274:2082060] [MT] PluginLoading: Required plug-in compatibility UUID 9F75337B-21B4-4ADC-B558-F9CADF7073A7 for plug-in at path '~/Library/Application Support/Developer/Shared/Xcode/Plug-ins/Unity4XC.xcplugin' not present in DVTPlugInCompatibilityUUIDs
2015-05-30 18:20:08.013 xcodebuild[2274:2082060] Failed to load plugin at: /Users/shyam/Library/Application Support/Developer/Shared/Xcode/Plug-ins/Unity4XC.xcplugin, skipping.  Reason for failure: *** -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]: attempt to insert nil object from objects[0]
  CXX(target) Release/obj.target/sharp/src/common.o
../src/common.cc:126:21: error: use of undeclared identifier 'atoi'
      orientation = atoi(&exif[0]);
                   ^
1 error generated.
make: *** [Release/obj.target/sharp/src/common.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/shyam/.nvm/v0.10.35/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at ChildProcess.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:820:12)
gyp ERR! System Darwin 14.3.0
gyp ERR! command "node" "/Users/shyam/.nvm/v0.10.35/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/shyam/Documents/_dev/recursive-image-search/node_modules/sharp
gyp ERR! node -v v0.10.35
gyp ERR! node-gyp -v v1.0.1
gyp ERR! not ok

npm ERR! sharp@0.10.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the sharp@0.10.0 install script.
npm ERR! This is most likely a problem with the sharp package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls sharp
npm ERR! There is likely additional logging output above.
npm ERR! System Darwin 14.3.0
npm ERR! command "/Users/shyam/.nvm/v0.10.35/bin/node" "/Users/shyam/.nvm/v0.10.35/bin/npm" "install" "sharp"
npm ERR! cwd /Users/shyam/Documents/_dev/recursive-image-search
npm ERR! node -v v0.10.35
npm ERR! npm -v 1.4.28
npm ERR! code ELIFECYCLE
npm ERR! not ok code 0
```

Cheers.

**Comment by lovell** [comment]
My best guess is that the C standard library, which provides `atoi`, is no longer made _automagically_ available to C++ source in the new version of clang/xcode. Adding `#include <cstdlib>` to `common.cc` should fix this - leave it with me.

**Comment by lovell** [comment]
Are you able to verify commit fa8f06f enables compilation with the latest Xcode by running:

```sh
npm install lovell/sharp
```

**Comment by doggan** [comment]
Verified! Compilation works great. Ty for the quick response.

Just for completeness, here is the console output. Seems like there are a few warnings, but I've confirmed that sharp is installed and running without any issues:

```bash
> sharp@0.10.1 install /Users/shyam/Documents/_dev/recursive-image-search/node_modules/sharp
> node-gyp rebuild

2015-05-31 12:46:00.637 xcodebuild[2524:75520] [MT] PluginLoading: Required plug-in compatibility UUID 9F75337B-21B4-4ADC-B558-F9CADF7073A7 for plug-in at path '~/Library/Application Support/Developer/Shared/Xcode/Plug-ins/Unity4XC.xcplugin' not present in DVTPlugInCompatibilityUUIDs
2015-05-31 12:46:00.638 xcodebuild[2524:75520] Failed to load plugin at: /Users/shyam/Library/Application Support/Developer/Shared/Xcode/Plug-ins/Unity4XC.xcplugin, skipping.  Reason for failure: *** -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]: attempt to insert nil object from objects[0]
  CXX(target) Release/obj.target/sharp/src/common.o
  CXX(target) Release/obj.target/sharp/src/utilities.o
  CXX(target) Release/obj.target/sharp/src/metadata.o
  CXX(target) Release/obj.target/sharp/src/resize.o
  CXX(target) Release/obj.target/sharp/src/sharp.o
  SOLINK_MODULE(target) Release/sharp.node
ld: warning: ld: warning: ignoring file /Library/Frameworks/Mono.framework/Versions/3.10.0/lib/libglib-2.0.dylib, file was built for i386 which is not the architecture being linked (x86_64): /Library/Frameworks/Mono.framework/Versions/3.10.0/lib/libglib-2.0.dylibignoring file /Library/Frameworks/Mono.framework/Versions/3.10.0/lib/libgobject-2.0.dylib, file was built for i386 which is not the architecture being linked (x86_64): /Library/Frameworks/Mono.framework/Versions/3.10.0/lib/libgobject-2.0.dylib

ld: warning: ignoring file /Library/Frameworks/Mono.framework/Versions/3.10.0/lib/libintl.dylib, file was built for i386 which is not the architecture being linked (x86_64): /Library/Frameworks/Mono.framework/Versions/3.10.0/lib/libintl.dylib
  SOLINK_MODULE(target) Release/sharp.node: Finished
sharp@0.10.1 node_modules/sharp
├── nan@1.8.4
├── bluebird@2.9.27
├── semver@4.3.5
└── color@0.8.0 (color-convert@0.5.2, color-string@0.3.0)
```

Thanks again.

**Comment by lovell** [comment]
Thanks for confirming and providing the original report. I suspect you won't be the only person to run into this so I'll be publishing this fix (and a couple of others) to npm as v0.10.1 shortly.

Running `brew reinstall glib` might fix the `libglib-2.0.dylib, file was built for i386` warnings, but I can't help with the `Unity4XC.xcplugin` warnings, sorry.
